query/stdlib/influxdata/influxdb/source.go

```diff
@@ -156,6 +157,84 @@ func createReadFilterSource(s plan.ProcedureSpec, id execute.DatasetID, a execut
 	), nil
 }
 
+type readGroupSource struct {
+	Source
+	reader   Reader
+	readSpec ReadGroupSpec
+}
+
+func ReadGroupSource(id execute.DatasetID, r Reader, readSpec ReadGroupSpec, alloc *memory.Allocator) execute.Source {
+	src := new(readGroupSource)
+
+	src.id = id
+	src.alloc = alloc
+
+	src.reader = r
+	src.readSpec = readSpec
+
+	src.runner = src
+	return src
+}
+
+func (s *readGroupSource) run(ctx context.Context) error {
+	stop := s.readSpec.Bounds.Stop
+	tables, err := s.reader.ReadGroup(
+		ctx,
+		s.readSpec,
+		s.alloc,
+	)
+	if err != nil {
+		return err
+	}
+	return s.processTables(ctx, tables, stop)
+}
+
+func createReadGroupSource(s plan.ProcedureSpec, id execute.DatasetID, a execute.Administration) (execute.Source, error) {
+	span, ctx := tracing.StartSpanFromContext(context.TODO())
```

Comment:
You could use `a.Context()` instead of `context.TODO()` here, although keeping contexts as fields in structs is discouraged, so it seems like `execute.Administration` shouldn't do that.  Ideally the context would be an argument to this function.